### ISSUE TITLE
latest-binaries.sh returns latest binaries without considering platform.

### ISSUE THREAD

**Issue by KrisJohnstone** [awslabs/amazon-eks-ami#1895]
**What happened**:
latest-binaries.sh returns the latest binary (as it says on the tin) but doesn't factor in the platform that will consume the binaries.

```
amazon-eks-ami/hack on  al2023-hardened-image-support [?] via  v3.12.4 on ☁️  product(us-west-2)
❯ ./latest-binaries.sh 1.29
kubernetes_version=1.29.6 kubernetes_build_date=2024-07-12
```

```
amazon-eks-ami/hack on  al2023-hardened-image-support [?] via  v3.12.4 on ☁️  product(us-west-2)
❯ aws s3 ls s3://amazon-eks/1.29.6/2024-07-12/bin/
                           PRE windows/
```

Which is fine, except that this being run on a linux box.

**What you expected to happen**:

Would expect the platform to be taken into account. i.e.

```
aws s3api list-objects-v2 --bucket amazon-eks --prefix "${MINOR_VERSION}" --query 'Contents[*].[Key]' --output text | cut -d'/' -f-2 | sort -Vru | head -n1
```

Would become:
```
aws s3api list-objects-v2 --bucket amazon-eks --prefix "${MINOR_VERSION}" --query 'Contents[*].[Key]' --output text | grep "${PLATFORM}" | cut -d'/' -f-2 | sort -Vru
```

Which would return:
```
amazon-eks-ami/hack on  al2023-hardened-image-support [?] via  v3.12.4 on ☁️  product-build (us-west-2)
❯ aws s3api list-objects-v2 --bucket amazon-eks --prefix "1.29" --query 'Contents[*].[Key]' --output text | grep linux | cut -d'/' -f-2 | sort -Vru
1.29.3/2024-04-19
1.29.0/2024-01-04
```

**How to reproduce it (as minimally and precisely as possible)**:
See above.

**Anything else we need to know?**:

**Environment**:
- AWS Region: us-west-2
- EKS Platform version: 1.29

**Comment by cartermckinnon** [comment]
This was fixed last week in #1889.

**Comment by KrisJohnstone** [comment]
Bah my search foo failed me. Apologies.
